src/header.rs

```diff
@@ -232,18 +229,17 @@ pub(crate) fn read_header(r: &mut dyn io::Read) -> io::Result<Value> {
 }
 
 fn parse_header_text_to_io_result(bytes: &[u8]) -> io::Result<Value> {
-    match parser::value(bytes) {
-        IResult::Done(remainder, header) => {
-            if !remainder.iter().all(|b| b" \t\n\r\0".contains(b)) {
-                return Err(invalid_data(format_args!("unexpected trailing data in header: {:?}", remainder)));
-            }
-            Ok(header)
-        },
-        IResult::Incomplete(needed) => {
-            Err(invalid_data(format_args!("could not parse Python expression: {:?}", needed)))
+    let without_newline = match bytes.split_last() {
+        Some((&b'\n', rest)) => rest,
+        Some(_) | None => return Err(invalid_data("missing newline")),
+    };
+    let as_str = std::str::from_utf8(without_newline).map_err(|_| invalid_data("could not parse utf-8"))?;
+    match as_str.parse() {
```

Comment:
Minor format nit:  Now that IResult is gone, this whole `match` can be replaced with `.map_err`.